src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -101,13 +101,17 @@ public Checksum CreateChecksum(AnalyzerReference reference, CancellationToken ca
                         WriteUnresolvedAnalyzerReferenceTo(unresolved, writer);
                         break;
 
+                    case AnalyzerReference analyzerReference when analyzerReference.GetType().FullName == "Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioUnresolvedAnalyzerReference":
```

Comment:
It's defined in `src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs`  Unfortunately, `UnresolvedAnalyzerReference` is sealed.

Definitely open to better suggestions here.